src/lib/fetch.ts

```diff
@@ -837,7 +839,7 @@ async function fetchActions(
           actions: currentActionList,
           hash: Ledger.fieldToBase58(Field(latestActionsHash)),
         });
-        currentAccountUpdateId = accountUpdateId;
```

Comment:
are we sure we want to remove this? if yes, why? in this `else` branch, we don't have `accountUpdateId === currentAccountUpdateId` so the line is not redundant

Reply:
Yes, sure, because the code after the end of this branch will also perform this operation. 
You can see the code comments below: 
```typescript
fetchedActions.forEach((fetchedAction) => {
    const { actionData } = fetchedAction;
    let latestActionsHash = Field(fetchedAction.actionState.actionStateTwo);
    let actionState = Field(fetchedAction.actionState.actionStateOne);

    if (actionData.length === 0)
      throw new Error(
        `No action data was found for the account ${publicKey} with the latest action state ${actionState}`
      );

    let { accountUpdateId: currentAccountUpdateId } = actionData[0];
    let currentActionList: string[][] = [];

    actionData.forEach((action, i) => {
      const { accountUpdateId, data } = action;
      const isLastAction = i === actionData.length - 1;
      const isSameAccountUpdate = accountUpdateId === currentAccountUpdateId;

      if (isSameAccountUpdate && !isLastAction) {
        currentActionList.push(data);
        currentAccountUpdateId = accountUpdateId;
        return;
      } else if (isSameAccountUpdate && isLastAction) {
        currentActionList.push(data);
      } else if (!isSameAccountUpdate && isLastAction) {
        // branch 3
        latestActionsHash = processActionData(
          currentActionList,
          latestActionsHash
        );
        actionsList.push({
          actions: currentActionList,
          hash: Ledger.fieldToBase58(Field(latestActionsHash)),
        });
        // The following line of code is redundant
        currentAccountUpdateId = accountUpdateId;
        currentActionList = [data];
      }

      latestActionsHash = processActionData(
        currentActionList,
        latestActionsHash
      );
      actionsList.push({
        actions: currentActionList,
        hash: Ledger.fieldToBase58(Field(latestActionsHash)),
      });
      // After the above branch 3 is executed, the currentAccoutUpdateId will also be updated here
      currentAccountUpdateId = accountUpdateId;
      currentActionList = [data];
    });
```

Reply:
thanks!